src/Domain/ShoppingItem.php

```diff
@@ -42,6 +42,17 @@ public function getId(): int
         return $this->id;
     }
 
+    /**
+     * Set the id of the shopping item.
+     *
+     * @param int $id
+     * @return int
+     */
+    public function setId(int $id): int
```

Comment:
Required: as discussed, can we fix the return type on this method? It either needs to be `void` or you need to `return $this`. If using the $this one, the return type ends up being `self` as that's how you indicate the the class returns an instance of itself.

Once this is fixed, we can merge this PR. Good job on the loop!

Reply:
Done!